### ISSUE TITLE
Push Docker images somewhere

### ISSUE THREAD

**Issue by jswny** [jacebrowning/memegen#726]
Would it be possible to push the Docker images for this repo somewhere? Since it's a public repo GitHub container registry would host them for free.

Since this project provides an API which I assume many others like myself want to bundle/deploy with a different application, the easiest way to do so would be to include the Docker image in a deployment. However, that's not possible currently since you'd have to build the image yourself, maybe using a submodule to pull down the code locally and then building it in CI and pushing it yourself.

If necessary I may have time to write the GitHub actions config required to do it.

**Comment by jacebrowning** [comment]
@jswny yeah, I'd be willing to consider that if you can provide a working setup with GitHub actions.

**Comment by Helvio88** [comment]
I have created my self-hosted Jenkins pipeline to periodically scan this repo for changes, build and publish to docker hub:
https://hub.docker.com/r/helvio/memegen

ARM and x64 images should be available. I was only able to test x64.

This is NOT official and won't be supported by @jacebrowning, but I'll try to help if I can.

**Comment by kyluca** [comment]
Hey @jacebrowning,

Just wondering if you're still interested in a solution to this? I'm looking for issues to work on for Hacktoberfest and wouldn't mind trying to setup GH Actions to publish an image to the GH container registry, but I realise this has been sidelined for over a year now. Is that still the brief?

Cheers,
Kyle

**Comment by jacebrowning** [comment]
@kyluca the solution proposed in https://github.com/jacebrowning/memegen/pull/801 looks pretty good if you want to try to bring that across the finish line. I left a comment about making the `docker push` optional.
